packages/backend/src/macadam.ts

```diff
@@ -0,0 +1,78 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import * as macadam from '@crc-org/macadam.js';
+import { macadamName } from './constants';
+
+interface StderrError extends Error {
+  stderr?: string;
+}
+export class MacadamVirtualMachine {
```

Comment:
The naming threw me off above because this isn't a virtual machine, this is a wrapper or handler class for the macadam API which could list multiple VMs. Maybe rename to something like `MacadamHandler` or `MacadamSupport`?